test/k23si/3SITxnTest.cpp

```diff
@@ -1794,10 +1822,1277 @@ seastar::future<> testScenario02() {
+                        K2INFO("SC07.prepare::OP_finalize_abort_k2. " << "status: " << status2.code << " with MESG: " << status2.message);
+                        K2EXPECT(status1, dto::K23SIStatus::OK);
+                        K2EXPECT(status2, dto::K23SIStatus::OK);
+                    });
+                })
+                .then([&] {
+                    return doInspectTxn(k1, mtr5, collname)
+                    .then([&](auto&& response)  {
+                        auto& [status, val] = response;
+                        K2INFO("SC07.case07::OP_inspect_txn_5. " << "status: " << status.code << " with MESG: " << status.message);
+                        K2INFO("State of txn_5: " << val.state);
+                        K2EXPECT(status, dto::K23SIStatus::OK);
+                        K2EXPECT(val.state, dto::TxnRecordState::InProgress);
+                    });
+                });
+            })
+            .then([&] {
+                K2INFO("------- SC07.case08 ( Using a transaction with newer timestamp to Push() an old one ) -------");
+                return doPush(k1, collname, mtr5, mtr6, ErrorCaseOpt::NoInjection)
+                .then([&](auto&& response)  {
+                    auto& [status, val] = response;
+                    K2INFO("SC07.case08::OP_Push_txn5_txn6(older as challenger). " << "status: " << status.code << " with MESG: " << status.message);
+                    K2INFO(val);
+                    K2EXPECT(status, dto::K23SIStatus::OK);
+                    K2EXPECT(val.winnerMTR, mtr6);
+                })
+                .then([&] {
+                    return seastar::when_all(doInspectTxn(k1, mtr5, collname), doInspectTxn(k1, mtr6, collname))
+                    .then([&](auto&& response) {
+                        auto& [resp1, resp2] = response;
+                        auto [status1, val1] = resp1.get0();
+                        auto [status2, val2] = resp2.get0();
+                        K2INFO("SC07.case08::OP_inspect_txn_5. " << "status: " << status1.code << " with MESG: " << status1.message);
+                        K2INFO("SC07.case08::OP_inspect_txn_6. " << "status: " << status2.code << " with MESG: " << status2.message);
+                        K2INFO("State of txn_5: " << val1.state);
+                        K2INFO("State of txn_6: " << val2.state);
+                        K2EXPECT(status1, dto::K23SIStatus::OK);
+                        K2EXPECT(status2, dto::K23SIStatus::KeyNotFound);
+                        K2EXPECT(val1.state, dto::TxnRecordState::ForceAborted);
+                        K2EXPECT(val2.state, dto::TxnRecordState::Created);
+                    });
+                });
+            });
+        }); // end do-with
+    }); // end sc-07
+}
+
+seastar::future<> testScenario08() {
+    std::cout << std::endl << std::endl;
+    K2INFO("+++++++ TestScenario 08: server-initiated txn abort +++++++");
```

Comment:
A transaction should only be in the forceAborted state if it was the incumbent in a PUSH operation and lost (or it failed heartbeat or retention window, but those are not relevant here). If a challenger transaction loses a PUSH, the client will see an AbortConflict but the transaction will be in the InProgress state and we rely on the client will END with an abort. If we don't see that then it is a bug in the code or the test. Here you are missing some END requests so the state of the previous test cases are not cleaned up. I commented on a couple but you can see if there are more.